nats.go

```diff
@@ -1083,14 +1086,11 @@ func (nc *Conn) processMsg(data []byte) {
 
 	sub.mu.Lock()
 
-	// This is a catch all for more than max messages delivered.
```

Comment:
You can't do that otherwise you would let messages accumulate to the channel, to a point probably of blocking down there, when trying to put it in the channel (since it is buffered).